packages/jbrowse-plugin-apollo/src/components/CreateApolloAnnotation.tsx

```diff
@@ -0,0 +1,295 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable unicorn/no-useless-undefined */
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import React, { useEffect, useMemo, useState } from 'react'
+
+import {
+  Box,
+  Button,
+  Checkbox,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  FormControlLabel,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+  Typography,
+} from '@mui/material'
+
+import { Dialog } from './Dialog'
+import { ApolloSessionModel } from '../session'
+import { AnnotationFeatureSnapshot } from '@apollo-annotation/mst'
+import { getRoot } from 'mobx-state-tree'
+import { ApolloRootModel } from '../types'
+import { ApolloInternetAccountModel } from '../ApolloInternetAccount/model'
+import { AddFeatureChange } from '@apollo-annotation/shared'
+import { Assembly } from '@jbrowse/core/assemblyManager/assembly'
+import { AbstractSessionModel } from '@jbrowse/core/util'
+
+interface CreateApolloAnnotationProps {
+  session: AbstractSessionModel
+  handleClose(): void
+  annotationFeature: AnnotationFeatureSnapshot
+  assembly: Assembly
+}
+
+// TODO: Integrate SO
+const isGeneOrTranscript = (annotationFeature: AnnotationFeatureSnapshot) => {
+  return (
+    annotationFeature.type === 'gene' ||
+    annotationFeature.type === 'mRNA' ||
+    annotationFeature.type === 'transcript'
+  )
+}
+
+// TODO: Integrate SO
+const isTranscript = (annotationFeature: AnnotationFeatureSnapshot) => {
+  return (
+    annotationFeature.type === 'mRNA' || annotationFeature.type === 'transcript'
+  )
+}
+
+export function CreateApolloAnnotation({
+  annotationFeature,
+  assembly,
+  handleClose,
+  session,
+}: CreateApolloAnnotationProps) {
+  const apolloSessionModel = session as unknown as ApolloSessionModel
+  const { internetAccounts } = getRoot<ApolloRootModel>(session)
+  const apolloInternetAccounts = internetAccounts.filter(
+    (ia) => ia.type === 'ApolloInternetAccount',
+  ) as ApolloInternetAccountModel[]
+  if (apolloInternetAccounts.length === 0) {
+    throw new Error('No Apollo internet account found')
+  }
+  const [apolloInternetAccount] = apolloInternetAccounts
+  const childIds = useMemo(
+    () => Object.keys(annotationFeature.children ?? {}),
+    [annotationFeature],
+  )
+
+  const [parentFeatureChecked, setParentFeatureChecked] = useState(true)
+  const [checkedChildrens, setCheckedChildrens] = useState<string[]>(childIds)
+  const [errorMessage, setErrorMessage] = useState('')
+  const [destinationFeatures, setDestinationFeatures] = useState<
+    AnnotationFeatureSnapshot[]
+  >([])
+  const [selectedDestinationFeature, setSelectedDestinationFeature] =
+    useState<AnnotationFeatureSnapshot>()
+
+  useEffect(() => {
```

Comment:
Instead of fetching features from the server, this should look at the already-fetched features stored in `session.apolloDataStore`.